### ISSUE TITLE
Multistat stopped showing data after Grafana update to 9.2.4 version

### ISSUE THREAD

**Issue by tomekso** [michaeldmoore/michaeldmoore-multistat-panel#70]
After the update of Grafana the Multistat stopped working. No errors - just do not show graph. In a "table view" data are available.

**Comment by michaeldmoore** [comment]
Ok, I'll check it out.

**Comment by prashsaka** [comment]
Thank you. By the way, this is the error that shows up in the console. And could be related.


![image](https://user-images.githubusercontent.com/67445443/224035072-d57061dd-e33b-4e4a-b1ba-697dcbf38294.png)

**Comment by benoitscherrer** [comment]
Same here

**Comment by SPKS-Berlin** [comment]
> Same here

**Comment by michaeldmoore** [comment]
The problem seems to be caused by a recent change in the D3 sort function.
I'm testing a fix and should get a release out in a few days time.  Thanks,
everyone for reporting this.

**Comment by benoitscherrer** [comment]
Hi,
There was a commit last month, with lots of code style updates, but also some actual code changes.
Any chance this is fixed?

Thank you!

**Comment by Pinkisagit** [comment]
It seems like d3.nest() method needs to change to d3.group() method.

**Comment by michaeldmoore** [comment]
Exactly.  I worked through the repercussions of this a while ago and
committed the changes to the github repo. The current code fails some lint
tests, so right now, it can't be set to an official releaseversion.  I've
been very busy with other life obligations, so I haven't gotten around to
filing up whatever the lint issues were a situation that is likely to
continue for some weeks still.

Right now, you can grab the latest commit from github and use that - it
seems to work fine.  You would have to configure Grafana to allow it as a
non-signed plugin too. That should work with the latest Grafana release.
Otherwise, hold off on the Grafana updates or find an alternative panel
plugin.

Sorry for all the messing about and slow responses.

**Comment by Pinkisagit** [comment]
I've just copied your latest code and it works, thanks mate. Let's hope you can sort of your lint issues. Good luck.

**Comment by prashanth-quantivly** [comment]
Is the latest code right as it comments out the use of `d3.nest`?

So, I updated the code as below.

```javascript
  // this.rows = d3
  //   .nest()
  //   .key(groupedLabelFunc)
  //   .rollup(function (d) {
  //     return d[0];
  //   })
  //   .entries(this.matchingRows)
  //   .forEach(function (x) {
  //     oo.push(x.value);
  //   });
  Array
    .from(d3.group(this.matchingRows, groupedLabelFunc))
    .map(d => {
      let key = d[0];
      let values = d[1];
      return { key: key, value: values[0] };
    })
    .forEach(x => oo.push(x.value));
```

Here is the **untested** updated file - https://gist.github.com/prashanth-quantivly/eacd7ea29b1b4501e263509cdaf6caca

**Comment by michaeldmoore** [comment]
Thanks Prashanth - your changes were very helpful.  They didn't quite work for the max/min/sum and mean aggregations, but based on your code, I got them all working correctly.

The current version (v1.7.4) is checked in here and should work properly.

I'm fighting the release actions mechanism right now - this signs the plugin after running a set of validation steps.  I don't understand why, but it seems grafana has changed - and broken - these github actions and I haven't figured out how to get it all working again.  It might take a while before I can get the signing/validation actions  working properly, so for the time bein, I can only offer this manual update process.  I'm going to be out of the country for the next month, so I don't expect anything to happen for a while.  Watch this space for updates...

**Comment by atsetogl** [comment]
Hello, could you guide me through the installation process of the v1.7.4 please? I am currently getting the plugin disabled because of missing signature.
